gateway/webrtc/client.js

```diff
@@ -318,42 +358,23 @@ function startChat() {
         },
         success: function(response) {
             console.log("Entering authorization success response handler", response);
-            handleVideo.iceConfig = [ response ];
-
-            loginPage.style.display = "none";
-
-            //errMessage.style.display = "none";
-            callPage.style.display = "block";
-            
-            callBtn.style.display='inline';
-            callToUsernameInput.style.display='inline';
-            hangUpBtn.style.display='none';
-            overlay.style.visibility='hidden';
 
-            $('#callToUsernameInput').focus();
+            //**********************
+            //Message listener can be safely registered
+            //**********************
+            registerMessageListenerCallback();
 
             //**********************
             //Starting a peer connection
             //**********************
-
-            //getting local video stream
-            navigator.mediaDevices.getUserMedia({
-              audio: true,
-              video: true
-            })
-            .then(handleVideo)
-            .catch(function(e) {
-              alert('getUserMedia() error: ' + e.name);
-            });
+            handleVideo(response, myStream);
             console.log("Exiting authorization success response handler");
         },
         error: function() {
             errMessage.style.display = "block";
```

Comment:
Since we are fiddling with the code in this area, it would be nice to also log the error that was returned. I.e. use function(xhr, status, error)  instead of function() and then console.log the error.